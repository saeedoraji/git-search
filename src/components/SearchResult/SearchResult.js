import React, { Component, lazy, Suspense } from "react";
import ExpansionPanel from "@material-ui/core/ExpansionPanel";
import ExpansionPanelDetails from "@material-ui/core/ExpansionPanelDetails";
import ExpansionPanelSummary from "@material-ui/core/ExpansionPanelSummary";
import ExpandMoreIcon from "@material-ui/icons/ExpandMore";

export const importRepos = (props) => lazy(() => import(`components/Repo`));

export class SearchResult extends Component {
  state = {
    expanded: false,
  };

  handleChangePanel = (panel) => async (_, isExpanded) => {
    const Repo = await importRepos();
    this.setState({ expanded: isExpanded ? panel : false, Repo, panel });
  };

  render() {
    const { expanded, Repo, panel } = this.state;
    const { users } = this.props;
    return (
      <>
        {users &&
          !!users.length &&
          users.map((user) => (
            <ExpansionPanel
              key={user.id}
              expanded={expanded === user.login}
              onChange={this.handleChangePanel(user.login)}
            >
              <ExpansionPanelSummary
                expandIcon={<ExpandMoreIcon />}
                aria-controls="panel1bh-content"
                id={user.id}
              >
                <span className="panel-text">{user.login}</span>
              </ExpansionPanelSummary>
              <ExpansionPanelDetails>
                <Suspense fallback={<div>Loading...</div>}>
                  {panel === user.login && Repo && (
                    <Repo username={user.login} />
                  )}
                </Suspense>
              </ExpansionPanelDetails>
            </ExpansionPanel>
          ))}
      </>
    );
  }
}
