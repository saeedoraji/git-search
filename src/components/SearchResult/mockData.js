export const mockData = [
  {
    login: "saeed",
    id: 37361,
    node_id: "MDQ6VXNlcjM3MzYx",
    avatar_url: "https://avatars2.githubusercontent.com/u/37361?v=4",
    gravatar_id: "",
    url: "https://api.github.com/users/saeed",
    html_url: "https://github.com/saeed",
    followers_url: "https://api.github.com/users/saeed/followers",
    following_url: "https://api.github.com/users/saeed/following{/other_user}",
    gists_url: "https://api.github.com/users/saeed/gists{/gist_id}",
    starred_url: "https://api.github.com/users/saeed/starred{/owner}{/repo}",
    subscriptions_url: "https://api.github.com/users/saeed/subscriptions",
    organizations_url: "https://api.github.com/users/saeed/orgs",
    repos_url: "https://api.github.com/users/saeed/repos",
    events_url: "https://api.github.com/users/saeed/events{/privacy}",
    received_events_url: "https://api.github.com/users/saeed/received_events",
    type: "User",
    site_admin: false,
    score: 1.0,
  },
  {
    login: "saeedsh92",
    id: 9940633,
    node_id: "MDQ6VXNlcjk5NDA2MzM=",
    avatar_url: "https://avatars0.githubusercontent.com/u/9940633?v=4",
    gravatar_id: "",
    url: "https://api.github.com/users/saeedsh92",
    html_url: "https://github.com/saeedsh92",
    followers_url: "https://api.github.com/users/saeedsh92/followers",
    following_url:
      "https://api.github.com/users/saeedsh92/following{/other_user}",
    gists_url: "https://api.github.com/users/saeedsh92/gists{/gist_id}",
    starred_url:
      "https://api.github.com/users/saeedsh92/starred{/owner}{/repo}",
    subscriptions_url: "https://api.github.com/users/saeedsh92/subscriptions",
    organizations_url: "https://api.github.com/users/saeedsh92/orgs",
    repos_url: "https://api.github.com/users/saeedsh92/repos",
    events_url: "https://api.github.com/users/saeedsh92/events{/privacy}",
    received_events_url:
      "https://api.github.com/users/saeedsh92/received_events",
    type: "User",
    site_admin: false,
    score: 1.0,
  },
];
