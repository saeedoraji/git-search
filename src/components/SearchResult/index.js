import { SearchResult } from "components/SearchResult/SearchResult";
export default SearchResult;
