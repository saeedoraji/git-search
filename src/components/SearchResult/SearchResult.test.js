import React, { Suspense } from "react";
import { mount } from "enzyme";
import { SearchResult, importRepos } from "./SearchResult";
import { create } from "react-test-renderer";
import { mockData } from "./mockData";

describe("<SearchResult />", () => {
  it("Render user list", () => {
    const wrapper = mount(<SearchResult users={mockData} />);
    expect(wrapper.find(".panel-text").length).toEqual(mockData.length);
  });

  it("Load Repos", async () => {
    const root = create(
      <Suspense fallback={<div>loading...</div>}>
        <SearchResult />
      </Suspense>
    );
    await importRepos;
    expect(root).toMatchSnapshot();
  });
});
