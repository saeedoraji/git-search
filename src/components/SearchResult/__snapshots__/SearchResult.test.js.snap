// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResult /> Load Repos 1`] = `null`;
